lib/manual_artefact_guarantor.rb

```diff
@@ -0,0 +1,140 @@
+require 'gds_api/content_store'
+
+class ManualArtefactGuarantor
+  def self.guarantee(manual_slug)
+    new(manual_slug).guarantee
+  end
+
+  attr_reader :manual_slug
+  def initialize(manual_slug)
+    @manual_slug = manual_slug
+  end
+
+  def guarantee
+    if !content_item_exists?
+      Guarantee.failure(manual_slug, reason: :does_not_exist)
+    elsif !content_item_is_a_manual?
+      Guarantee.failure(manual_slug, reason: :is_not_a_manual)
+    elsif artefact_exists?
+      handle_existing_artefact
+    else
+      handle_missing_artefact
+    end
+  end
+
+  def content_item
+    # Do this to allow memoizing the nils we get from a 404
+    unless defined? @content_item
+      @content_item = fetch_content_item
+    end
+    @content_item
+  end
+
+  def artefact
+    unless defined? @artefact
+      @artefact = fetch_artefact
+    end
+    @artefact
+  end
+
+  private
```

Comment:
This should be outdented as per the styleguide: https://github.com/alphagov/styleguides/blob/master/ruby.md#classes